### DISCUSSION TITLE
High memory usage

### DISCUSSION THREAD

**Discussion by drmargarit** [Ross-Patterson/Portals-Desktop-Organization#156] [General]
I have 4 portals on my desktop. Memory usage is 780M.  Is that normal? How can I reduce it?

**Comment by Ross-Patterson** [comment]
This is definitely not normal, how many files per portal?

**Comment by drmargarit** [comment]
One of the folders has 33000+ files in it while another has 1700 files.  Apparently, it is best to create folders full of shortcuts and then create a portal for that folder. It isn't ideal because it adds another layer of clicking to open the folder but the memory usage is down to 200+M.

**Comment by Ross-Patterson** [comment]
@drmargarit try in the new beta version 3.2.0.2, memory usage should be massively reduced. You can switch to the beta channel in the settings.

Let me know how it goes.

**Reply by githubmk5** [reply]
[portals-memory.pdf](https://github.com/user-attachments/files/23148263/portals-memory.pdf)
Hello, I found out that that there is a huge discrepancy between the amount of memory portals needs and the amount of memory it reserves. I always looked in Taskmanager -->Processes and after a fresh start of portals and a longer time afterwards portals uses a lot of memory and then goes down to a reasonable amount of memory. But when you go to Taskmanger -->Details and sort after "committed size" (memory), portals is (always!) the biggest one, even larger then the process of a running hyper-v virtual machine. Could this be reduced? In the processes-view you see Portals using 9,6MB memory. I think I forced it to reduce to that amount by starting a different big process claiming a lot of memory several times until it really startet but portals is still running, so it doesn't need that much memory? Thank you for your effort

**Comment by githubmk5** [comment]
Hello, I have 5 Portals on my desktop with 22 Tabs containing 200 Files, counting 75MB altogether, mostly Shortcuts. 

For months, my Taskmanger has shown a memory consumption of over 1GB most of the time but later the day this can reduce to between 100 and 200MB. I don't know how to reproduce this but today I was on 112MB only after 30 Minutes. 

And what I at least did, was to click all tabs on portals in order to count the files (which can easier be done by right clicking the main folder of all portal-folders..) and to check, which Portals-Version I am on and do an update search. Maybe these actions changed something? 
I just repeated this after a Portals restart, this time memory comsumtion stays at 1,1GB. Not reproducable.

I have been on 3.3.0.9 for a long time now. I am on Windows 11 Pro, very recent Laptop with Core 7 150U and 16GB RAM

By the way, I love Portals, it helps me so much to stay oriented and tidy all the time. Thank you very much for this wonderful peace of software.

One more little issue I have is, that 2 of the 200 Shortcuts lose their icon right after program start and show a red X. If I drag these Shortcuts from their Tabs to the desktop an back, they reappear

**Reply by githubmk5** [reply]
any comment? And I also sometimes have this little rectangle in the upper left corner mentioned by someone else
